DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/Delegates/DelegateCoursesControllerTests.cs

```diff
@@ -0,0 +1,179 @@
+    using FluentAssertions;
+    using FluentAssertions.Execution;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+    using NUnit.Framework;
+
+    public class DelegateCoursesControllerTests
+    {
+        private readonly IEnumerable<ApplicationDetails> applicationOptions =
+            Builder<ApplicationDetails>.CreateListOfSize(1).Build();
+
+        private readonly IEnumerable<CourseStatisticsWithAdminFieldResponseCounts> courses =
+            Builder<CourseStatisticsWithAdminFieldResponseCounts>.CreateListOfSize(1).Build();
+
+        private readonly IEnumerable<Category> categories = Builder<Category>.CreateListOfSize(2).Build();
+
+        private readonly IEnumerable<Topic> topics = Builder<Topic>.CreateListOfSize(2).Build();
+
+        private DelegateCoursesController controller = null!;
+        private DelegateCoursesController controllerWithCookies = null!;
+        private ICourseCategoriesDataService courseCategoryDataService = null!;
+        private ICourseService courseService = null!;
+        private ICourseTopicsDataService courseTopicsDataService = null!;
+        private HttpRequest httpRequest = null!;
+        private HttpResponse httpResponse = null!;
+
+        [SetUp]
+        public void Setup()
+        {
+            courseCategoryDataService = A.Fake<ICourseCategoriesDataService>();
+            courseTopicsDataService = A.Fake<ICourseTopicsDataService>();
+            courseService = A.Fake<ICourseService>();
+
+            A.CallTo(
+                () => courseService.GetCentreSpecificCourseStatisticsWithAdminFieldResponseCounts(A<int>._, A<int>._)
+            ).Returns(courses);
+            A.CallTo(() => courseCategoryDataService.GetCategoriesForCentreAndCentrallyManagedCourses(A<int>._))
+                .Returns(categories);
+            A.CallTo(() => courseTopicsDataService.GetCourseTopicsAvailableAtCentre(A<int>._)).Returns(topics);
+            A.CallTo(
+                () => courseService.GetApplicationOptionsAlphabeticalListForCentre(A<int>._, A<int?>._, A<int?>._)
+            ).Returns(applicationOptions);
+
+            httpRequest = A.Fake<HttpRequest>();
+            httpResponse = A.Fake<HttpResponse>();
+
+            controller = new DelegateCoursesController(
+                    courseService,
+                    courseCategoryDataService,
+                    courseTopicsDataService
+                )
+                .WithDefaultContext()
+                .WithMockUser(true, 101)
+                .WithMockTempData();
+
+            const string cookieName = "DelegateCoursesFilter";
+            const string cookieValue = "Status|Active|false";
+
+            controllerWithCookies = new DelegateCoursesController(
+                    courseService,
+                    courseCategoryDataService,
+                    courseTopicsDataService
+                )
+                .WithMockHttpContext(httpRequest, cookieName, cookieValue, httpResponse)

```

Comment:
Just to clarify, on the old PR, Steve had a question about setting up the cookies inside the tests to make the individual tests read slightly easier. Was this decided against?

Reply:
It proved to be a bit of pain to implement and considering its required in a good number of the tests it made sense to leave it where it is.